contracts/contracts/coordination/SigningCoordinator.sol

```diff
@@ -292,40 +293,62 @@ contract SigningCoordinator is Initializable, AccessControlDefaultAdminRulesUpgr
             }
         }
         require(chainDeployed, "Not already deployed");
-        ServerAdmin storage serverAdmin = signingCohort.serverAdminConditions[msg.sender];
+    }
+
+    function setSigningCohortConditions(
+        uint32 cohortId,
+        uint256 chainId,
+        bytes calldata conditions
+    ) external {
+        SigningCohort storage signingCohort = signingCohorts[cohortId];
+        conditionsCheck(signingCohort, chainId);
+        require(
+            signingCohort.authority == msg.sender,
+            "Only the cohort authority can set conditions"
+        );
+        signingCohort.conditions[chainId] = conditions;
+        emit SigningCohortConditionsSet(cohortId, msg.sender, chainId, conditions);
+    }
+
+    function setSigningCohortConditions(
+        uint32 cohortId,
+        uint256 chainId,
+        bytes calldata conditions,
+        uint256 serverId
+    ) external {
+        SigningCohort storage signingCohort = signingCohorts[cohortId];
+        conditionsCheck(signingCohort, chainId);
+        require(serverId > 0, "Server id can't be zero");
+        Server storage server = signingCohort.serverConditions[serverId];
 
         require(
-            signingCohort.authority == msg.sender || serverAdmin.active,
+            signingCohort.authority == msg.sender || server.activeServerAdmins[msg.sender],
```

Comment:
Does this mean that the cohort authority can override a server admin for setting the serverId-specific condition? If so, I don't think we want to give the cohort authority that power.

Only the server admin can set the serverId-specific condition. (cc @arjunhassard )

Reply:
But cohort admin can set itself to server admin anyway, right?

Reply:
Removed privilege for cohort admin